docs/database-processes.md

```diff
@@ -12,6 +12,34 @@ failed to lock storage
 Unhandled exception. System.Exception: ScErrCantStartDatabase (SCERR10004): Attempting to start the database failed. Failed to start ConsoleApp. Data process reported ScErrDbAlreadyStarted (SCERR2108): A Starcounter database with the specified name is already started.
 ```
 
+Starcounter tooling or management components control `scdata` processes and their lifetime. Under normal circumstances no manual interaction with the `scdata` process shall be required from Starcounter end users.
+
+### Deciding if a database process is started
+
+When a request to start a database comes, Starcounter starts corresponding database process if it's not running. A client, such as an application asking to connect to Starcounter, specifies the path to the database directory, and Starcounter combines this with an internal, opaque database engine version number to determine if the process is running.
+
+Let's look at two examples:
+
+An application connects to Starcounter:
+* Application asks to connect to `./mydb`, i.e `Database=./mydb`.
+* Starcounter checks the internal version of Starcounter binaries referenced by the **application**, e.g. "3.2.1".
+* Starcounter combines the two above values to form a unique database identity, and use that to check if the database process is running.
+
+Starcounter tool starts a database:
+* Tooling is used to start the database i.e. `dotnet start start ./mydb`.
```

Comment:
```suggestion
* The Starcounter command line interface (CLI) is used to start the database `mydb`. This can be done with the command `dotnet star start ./mydb`.
```